packages/tools/kda-cli/src/components/execute.tsx

```diff
@@ -0,0 +1,31 @@
+import { IAnswers, IQuestion } from '../questions/questions.js';
+
+import { Spinner } from '@inkjs/ui';
+import { Text } from 'ink';
+import React, { useEffect, useState } from 'react';
+
+export const Execute = ({
+  message,
+  action,
+  answers,
+  onSubmit,
+}: IQuestion & {
+  answers: IAnswers;
+  onSubmit: (v: string | string[]) => void;
+}): ReturnType<typeof Spinner> => {
+  const [error, setError] = useState<string>();
+  useEffect(() => {
+    if (action === undefined) {
+      return setError('No action provided');
```

Comment:
`setError` is returning a void, so they do effectively the same. Though if for some reason `setError` changes signature, it would indeed introduce some incorrect behavior.